storage/reader.go

```diff
@@ -439,3 +440,9 @@ func (mrd *MultiRangeDownloader) Wait() {
 func (mrd *MultiRangeDownloader) GetHandle() []byte {
 	return mrd.reader.getHandle()
 }
+
+// CurrentState returns the current state of the MultiRangeDownloader.
+// It returns true if the stream is still open and false if it is closed.
+func (mrd *MultiRangeDownloader) CurrentState() bool {
```

Comment:
Maybe this should return an error rather than a bool?

I would expect:
* nil error = still open
* sentinel error (maybe call it `ErrStreamClosed`) if there was a graceful closure with all work completed.
* error that was returned by `MultiRangeDownloader.Close` otherwise.

That seems like it would give the most context to the caller.

Reply:
Actually, it looks like `close` always returns a nil error. I am also confused as to why we are returning only the `MultiRangeDownloader.Close` (are there no errors elsewhere [maybe the callback] that would be relevant? 

I am also unsure of how to detect a graceful closure (if there is no `activeTask` maybe?). @shubham-diwakar you probably know how to do that better than me, what do you think?